api/src/index.js

```diff
@@ -11,16 +11,95 @@ app.use(bodyParser.json());
 
 const apiRouter = express.Router();
 
-// You can delete this route once you add your own routes
-apiRouter.get("/", async (req, res) => {
-  const SHOW_TABLES_QUERY =
-    process.env.DB_CLIENT === "pg"
-      ? "SELECT * FROM pg_catalog.pg_tables;"
-      : "SHOW TABLES;";
-  const tables = await knex.raw(SHOW_TABLES_QUERY);
-  res.json({ tables });
+//future-meals
+apiRouter.get("/future-meals", async(req, res) => {
+try {
+  //Get the current day and time
+  const currentDateTime = new Date();
+
+  //Query all the meals where "when" is in the future
+  const futureMeals = await knex('Meal').where('when_meal', '>', currentDateTime);
+
+  //Send the future meals a JSON responde
+  res.json(futureMeals);
+} catch (error) {
+  console.error("Error fetching future meals:", error);
+  res.status(500).json({error: "Internal Server Error"});
+  }
+});
+
+//past-meals
+apiRouter.get("/past-meals", async(req,res) => {
+try{
+  const currentDateTime = new Date();
+  const pastMeals = await knex('Meal').where('when_meal', '<', currentDateTime);
+
+  res.json(pastMeals);
+} catch(error) {
+  console.error("Error fetching past meals:", error);
+  res.status(500).json({error: "Internal Server Error"});
+}
+});
+
+//all-meals	Respond with all meals sorted by ID
+
+apiRouter.get("/all-meals", async(req,res) => {
+try {
+  const allMeals = await knex('Meal')
+  .select('*')
+  .orderBy('id', 'asc');
+  res.json(allMeals);
+} catch(error) {
+  console.error("Error fetching past meals:", error);
+  res.status(500).json({error: "Internal Server Error"});
+}
+
 });
 
+//First-meal
+apiRouter.get("/first-meal", async(req,res) => {
+  try {
+    const firstMeal = await knex('Meal')
+    .select('*')
+    .orderBy('id', 'asc')
+    .first();
+    res.json(firstMeal);
+  } catch(error) {
+    console.error("Error fetching past meals:", error);
+    res.status(500).json({error: "Internal Server Error"});
+  }
+  
+  });
+
+  //Last-meal
+apiRouter.get("/last-meal", async(req,res) => {
+  try {
+    const lastMeal = await knex('Meal')
+    .select('*')
+    .orderBy('id', 'desc')
+    .first();
+    if (lastMeal) {
+      res.json(lastMeal);
+    } else {
+      res.status(404).json({ error: "No meals found" });
+    }
+  } catch(error) {
+    console.error("Error fetching past meals:", error);
+    res.status(500).json({error: "Internal Server Error"});
+  }
+  
+  });
+
+// apiRouter.get("/future-meals", async (req, res) => {
```

Comment:
It's better not to leave commented code in the code that is being committed to the repository :)